toqito/nonlocal_games/binary_constraint_system_game.py

```diff
@@ -0,0 +1,184 @@
+import numpy as np
+import networkx as nx
+from toqito.nonlocal_games.nonlocal_game import NonlocalGame
+from typing import List
+
+def create_bcs_constraints(M: np.ndarray, b: np.ndarray) -> List[np.ndarray]:
+    r"""
+    Construct a list of constraints for a binary constraint system (BCS) game.
+
+    Each row of ``M`` is concatenated with the value :math:`(-1)^{b[i]}` to
+    form a 1D NumPy array. The resulting list of arrays can then be passed
+    to :meth:`toqito.nonlocal_games.nonlocal_game.NonlocalGame.from_bcs_game`.
+
+    This construction is based on binary constraint system games, which are
+    closely related to the linear system games studied in :cite:`cleve2017perfect`.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        import numpy as np
+        from binary_constraint_system_game import create_bcs_constraints
+        from toqito.nonlocal_games.nonlocal_game import NonlocalGame
+
+        M = np.array([[1, 1], [1, 1]], dtype=int)
+        b = np.array([0, 1], dtype=int)
+        constraints = create_bcs_constraints(M, b)
+
+        game = NonlocalGame.from_bcs_game(constraints)
+
+    References
+    ----------
+    .. bibliography::
+       :filter: docname in docnames
+
+    :param M: A binary matrix of shape ``(m, n)`` specifying which variables appear
+              in each of the ``m`` constraints. Each entry is either 0 or 1.
+    :param b: A binary vector of length ``m``. For each row ``i``, the exponent
+              :math:`b[i]` determines whether the right-hand side is +1 or -1,
+              via :math:`(-1)^{b[i]}`.
+    :return: A list of 1D NumPy arrays, where each array is of length ``n+1``.
+             The first ``n`` entries correspond to a row of ``M``, and the last
+             entry is :math:`\pm 1` depending on ``b[i]``.
+    :rtype: List[np.ndarray]
```

Comment:
Please do not put in explicit return types. We know what these should be based on the type-hinting.